common/transport/snmp/src/main/java/org/thingsboard/server/transport/snmp/service/SnmpTransportService.java

```diff
@@ -183,19 +183,29 @@ public void onToDeviceRpcRequest(DeviceSessionContext sessionContext, TransportP
         SnmpMethod snmpMethod = SnmpMethod.valueOf(toDeviceRpcRequestMsg.getMethodName());
         JsonObject params = JsonConverter.parse(toDeviceRpcRequestMsg.getParams()).getAsJsonObject();
 
-        String oid = Optional.ofNullable(params.get("oid")).map(JsonElement::getAsString).orElse(null);
+        String key = Optional.ofNullable(params.get("key")).map(JsonElement::getAsString).orElse(null);
         String value = Optional.ofNullable(params.get("value")).map(JsonElement::getAsString).orElse(null);
-        DataType dataType = Optional.ofNullable(params.get("dataType")).map(e -> DataType.valueOf(e.getAsString())).orElse(DataType.STRING);
 
-        if (oid == null || oid.isEmpty()) {
-            throw new IllegalArgumentException("OID in to-device RPC request is not specified");
-        }
         if (value == null && snmpMethod == SnmpMethod.SET) {
             throw new IllegalArgumentException("Value must be specified for SNMP method 'SET'");
         }
 
-        PDU request = pduService.createSingleVariablePdu(sessionContext, snmpMethod, oid, value, dataType);
-        sendRequest(sessionContext, request, new RequestInfo(toDeviceRpcRequestMsg.getRequestId(), SnmpCommunicationSpec.TO_DEVICE_RPC_REQUEST));
+        sessionContext.getProfileTransportConfiguration().getCommunicationConfigs().stream()
+                .filter(communicationConfig -> communicationConfig.getSpec() == SnmpCommunicationSpec.TO_DEVICE_RPC_REQUEST)
+                .findFirst()
+                .ifPresent(communicationConfig -> {
```

Comment:
What if it is not present? Then we should reply with the error and not simply ignore the request!